scripts/install.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/sh
+
+set -u
+
+BYTEBASE_VERSION="1.1.1"
+readonly BYTEBASE_VERSION
+
+abort() {
+    printf "%s\n" "$@" >&2
+    exit 1
+}
+
+uname_os() {
+    local OS="$(uname -s)"
+    if [[ "${OS}" != "Darwin" && "${OS}" ]]; then
+        abort "OS ${OS} is not support, bytebase is only supported on Linux and MacOS"
+    fi
+    echo ${OS}
+}
+
+uname_arch() {
+    local ARCH=$(uname -m)
+    if [[ "${ARCH}" == "amd64" || "${ARCH}" == "x86_64" ]]; then
+        ARCH="x86_64"
+    elif [[ "${ARCH}" != "arm64" ]]; then
+        abort "Arch ${ARCH} is not support, bytebase is only supported on x86_64, amd64 and arm64"
+    fi
+    echo ${ARCH}
+}
+
+test_curl() {
+    local curl_version=$(curl --version 2>/dev/null)
+    if [ $? -ne 0 ]; then
+        abort "You must install curl before installing bytebase."
+    fi
+}
+
+test_tar() {
+    local tar_version=$(tar --version 2>/dev/null)
+    if [ $? -ne 0 ]; then
+        abort "You must install tar before installing bytebase."
+    fi
+}
+
+http_download() {
+    local local_file=$1
+    local source_url=$2
+
+    echo "Start downloading bytebase version ${BYTEBASE_VERSION} from ${source_url}..."
+
+    local code=$(curl -w '%{http_code}' -L -o "${local_file}" "${source_url}")
+    if [ "$code" != "200" ]; then
+        abort "Failed to download bytebase version ${BYTEBASE_VERSION} from ${source_url}, status code: ${code}"
+    fi
+
+    echo "Completed downloading bytebase version ${BYTEBASE_VERSION}"
+}
+
+execute() {
+    OS="$(uname_os)"
+    echo "OS: ${OS}"
+    ARCH="$(uname_arch)"
+    echo "ARCH: ${ARCH}"
+
+    test_curl
+    test_tar
+
+    # Initial bytebase direcoty
+    bytebase_dir="/opt/bytebase"
+    (sudo mkdir -p "${bytebase_dir}") || abort "cannot create directory ${bytebase_dir}"
+
+    install_dir="/usr/local/bin"
+
+    tmp_dir=$(mktemp -d) || abort "cannot create temp directory"
+    # Clean the tmpdir automatically if the shell script exit
+    trap "rm -r ${tmp_dir}" EXIT
+
+    echo "Downloading tarball into ${tmp_dir}"
+    tarball_name="bytebase_${BYTEBASE_VERSION}_${OS}_${ARCH}.tar.gz"
+    http_download "${tmp_dir}/${tarball_name}" \
+        "https://github.com/bytebase/bytebase/releases/download/${BYTEBASE_VERSION}/${tarball_name}"
+
+    echo "Start extracting tarball into ${bytebase_dir}..."
+    cd "${bytebase_dir}" && sudo tar -xzf "${tmp_dir}/${tarball_name}"
+
+    echo "Start installing bytebase and bb"
+    sudo install "${bytebase_dir}/bytebase" "${install_dir}"
+    echo "Installed bytebase to ${install_dir}"
```

Comment:
mention version in bytebase and bb echo?

Reply:
SG.